packages/multichain-account-service/src/MultichainAccountWallet.ts

```diff
@@ -326,4 +326,101 @@ export class MultichainAccountWallet<
       await group.align();
     }
   }
+
+  /**
+   * Discover and create accounts for all providers.
+   *
+   * NOTE: This method should only be called on a newly created wallet.
+   *
+   * @param opts - The options for the discovery and creation of accounts.
+   * @param opts.skippedProviders - The providers to skip.
+   * @returns The accounts for each provider.
+   */
+  async discoverAndCreateAccounts({
+    skippedProviders = [],
+  }: {
+    skippedProviders: AccountProviderType[];
+  }): Promise<{
+    [AccountProviderType]: Bip44Account<KeyringAccount>[];
+  }> {
+    const providers = this.#providers.filter(
+      (provider) => !skippedProviders.includes(provider.providerType),
+    );
+
+    const results: Record<AccountProviderType, Bip44Account<KeyringAccount>[]> =
+      {};
+    const nextIndex = new Map<Bip44Provider, number>(
+      providers.map((p) => [p, 0]),
+    );
+    const stopped = new Set<Bip44Provider>();
+    const inFlight = new Map<Bip44Provider, Promise<void>>();
+
+    let high = 0;
+
+    const schedule = (p: Bip44Provider, index: number) => {
+      const run = (async () => {
+        try {
+          const accounts = await p.discoverAndCreateAccounts({
+            entropySource: this.#entropySource,
+            groupIndex: index,
+          });
+
+          inFlight.delete(p);
+
+          if (accounts.length > 0) {
```

Comment:
It should be `!accounts.length` here no? We stop the discovery if we don't find any accounts for a given index I'd say?